### PR TITLE
GH-15791 UpliftDRF implement early stopping

### PR DESCRIPTION
**Pull request by maurever**
Part of issue: #15791

### CODE REVIEW

**Review by wendycwong — DISMISSED** [review]

**Review by wendycwong — APPROVED** [review]

File: h2o-py/tests/testdir_algos/uplift/pyunit_uplift_rf_early_stopping.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    uplift_model_es.train(y=response_column, x=x_names, training_frame=train_h2o, validation_frame=valid_h2o)
+    print(uplift_model_es)
+    
+    num_trees = pyunit_utils.extract_from_twoDimTable(uplift_model._model_json["output"]["model_summary"],
+                                                     "number_of_trees", takeFirst=True)
+    num_trees_es = pyunit_utils.extract_from_twoDimTable(uplift_model_es._model_json["output"]["model_summary"],
+                                                          "number_of_trees", takeFirst=True)
+    print("Number of tress built with early stopping: {0}.  Number of trees built without early stopping: "
+          "{1}".format(num_trees_es[0], num_trees[0]))
+    assert num_trees_es[0] <= num_trees[0], "Early stopping criteria AUUC is not working."
```
Comment (valenad1) [inline h2o-py/tests/testdir_algos/uplift/pyunit_uplift_rf_early_stopping.py:63]: IMHO if early stopping is tested correctly and deterministically we have to assert `num_trees_es[0] < num_trees[0]`

I have added test for `uplift_model_es.actual_params["ntrees"]` there was a bug with this parametr before. So to be sure its also correctly setup here.

File: h2o-r/tests/testdir_algos/uplift/runit_uplift_rf_early_stopping.R [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    print(model_es)
+
+    num_trees <- model@model$model_summary$number_of_trees
+    print("Number of trees built without AUUC early-stopping:")
+    print(num_trees)
+    num_trees_es <- model_es@model$model_summary$number_of_trees
+    print("Number of trees built with AUUC early-stopping:")
+    print(num_trees_es)
+    expect_true(num_trees >= num_trees_es)
```
Comment (valenad1) [inline h2o-r/tests/testdir_algos/uplift/runit_uplift_rf_early_stopping.R:67]: Same here.

File: h2o-py/tests/testdir_algos/uplift/pyunit_uplift_rf_early_stopping.py (hunk 1 above)
Comment (maurever) [reply h2o-py/tests/testdir_algos/uplift/pyunit_uplift_rf_early_stopping.py:63]: Makes sense. Thanks!

**Review by valenad1 — APPROVED** [review]
Thank you!

**Pull request merged**